tests/units/badge/minter/test_claim.cairo

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: MIT
+
+%lang starknet
+
+// Starkware dependencies
+from starkware.cairo.common.cairo_builtins import HashBuiltin, SignatureBuiltin
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.math import assert_not_zero
+from starkware.cairo.common.uint256 import Uint256
+
+// Local dependencies
+from tests.units.badge.minter.library import setup, prepare
+from src.badge.minter import claim
+
+@view
+func __setup__{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() {
+    return setup();
+}
+
+@external
+func test_claim_nominal_case{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, ecdsa_ptr : SignatureBuiltin*
+}() {
+    alloc_locals;
+
+    // Extract context variables
+    local minter_owner: felt;
+    local proxy_admin: felt;
+    local carbonable_badge_contract_address: felt;
+    local public_key: felt;
+    local sig0: felt;
+    local sig1: felt;
+    local badge_type: felt;
+    %{ 
+        ids.minter_owner = context.mocks.owner
+        ids.proxy_admin = context.mocks.proxy_admin
+        ids.carbonable_badge_contract_address = context.mocks.carbonable_badge_contract_address
+        ids.public_key = context.whitelist.public_key
+        ids.sig0 = context.whitelist.sig0
+        ids.sig1 = context.whitelist.sig1
+        ids.badge_type = context.whitelist.badge_type
+    %}
+
+    // Prepare minter instance
+    prepare(minter_owner, public_key, carbonable_badge_contract_address, proxy_admin);
+
+    // Claim badge
+    %{ stop_prank_callable = start_prank(context.whitelist.whitelisted_user_address) %}
+    %{ mock_call(context.mocks.carbonable_badge_contract_address, "mint", [0]) %}
+    %{ mock_call(context.mocks.carbonable_badge_contract_address, "balanceOf", [0, 0]) %}
```

Comment:
Why, for a nominal case, the `mint` and `balanceOf` are to 0 ?

Reply:
For the mint it is a mistake on my part, it should just be []. For the balance, it is necessary to have not mint the NFT (and thus to have a balance of 0) for mint. I'll do an extra commit to fix the first point and add an explicative comment for the balance.